src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java

```diff
@@ -227,31 +222,22 @@ private void generateTwoCommitDiffPatch(RevCommit commitOld, RevCommit commitNew
 
     private void generateDiffPatchWithGitCommand(int headNum, String patchFormat) throws Exception {
         if ("show".equals(patchFormat)) {
-            final List<String> showCmds = new ArrayList<>();
-            showCmds.add("sh");
-            showCmds.add("-c");
-            showCmds.add("git show > show.patch");
-            final Process process = new ProcessBuilder()
-                    .directory(new File(repoPath))
-                    .command(showCmds)
-                    .start();
-            final int code = process.waitFor();
-            if (code != 0) {
-                throw new IllegalStateException(
-                        "an error occurred when running git show > show.patch");
-            }
+            runShellCommand("git show > show.patch");
         }
-        else {
+        else if ("diff".equals(patchFormat)) {
+            runShellCommand("git diff HEAD~1 HEAD > show.patch");
+        }
+        else if ("format".equals(patchFormat)) {
             final Process process = new ProcessBuilder()
                     .directory(new File(repoPath))
                     .command(GIT, "format-patch", "-1")
                     .inheritIO()
                     .start();
             final int code = process.waitFor();
-            if (code != 0) {
-                throw new IllegalStateException(
-                        "an error occurred when running git format-patch -1");
-            }
+            stateException(code, "an error occurred when running git format-patch -1");
```

Comment:
can you use `runShellCommand("git format-patch -1")` here?

Reply:
done